### ISSUE TITLE
One-Step Publish for Meltano and SDK

### ISSUE THREAD

**Issue by MeltyBot** [meltano/meltano#2978]
Migrated from GitLab: https://gitlab.com/meltano/meltano/-/issues/3044

Originally created by @aaronsteers on 2021-11-05 15:57:36

---

We have a one-click prerelease publish in the SDK. That's a simpler problem because we don't have release notes and the version number is just an auto-generated `dev` bump using the CI build number.
 
Opening this issue to figure out how we get to similar simplicity for normal deployments.

## Goals

1. Continue frequent releases.
2. Reduce cost to dev productivity.
3. Make dual-channel releases (`edge`/`main`) more viable, as proposed in https://gitlab.com/meltano/meta/-/issues/135

## Statu Quo

Here are the current manual deployments steps (excluding marketing/slack/tweets, etc.):

1. Open a release branch. (Triggers an auto-commit with the version bump.)
2. Open a tag from the release branch. (Triggers the PyPi publish.)
3. Groom the changelog. (Can be performed before, after, or in parallel with the above - since the changelog itself is not shipped.)

### Changelog automation

I don't know as much about this space but I believe there are some available options.

### One-Step version bump, validation, and publish

Rather than requiring the tag to be created explicitly in a separate step, we can modify the CI pipeline to _also_ run the publish step in the special CI job that runs when a new branch is created like `release/*`.

The special CI pipeline for `release/*` tags can be removed in favor of the branch pipeline, to eliminate the need for a second step, and avoid "chaining" of auto-commits and CI jobs. We could still create the tag as part of the CI pipeline, rather than as a manual step.

### Proposed flow

Releases:

1. As release manager, I create a correctly-named release branch which triggers the release pipeline.
2. I click "Merge when pipeline succeeds" immediately - meaning, a successful CI run (including publish steps) will close out the version bump.

Change Log Grooming:

1. If needed, I create and merge a _separate_ changelog grooming MR.

**Comment by MeltyBot** [comment]
[View 9 previous comments from the original issue on GitLab](https://gitlab.com/meltano/meltano/-/issues/3044)

**Comment by tayloramurphy** [comment]
@WillDaSilva @kgpayne is this still relevant?

**Comment by WillDaSilva** [comment]
Superseded by:
- https://github.com/meltano/internal-tech-ops/issues/11
